data/transaction/status.go

```diff
@@ -33,101 +33,120 @@ func (tx TxStatus) String() string {
 	return string(tx)
 }
 
-// StatusComputer computes a transaction status
-type StatusComputer struct {
-	SelfShard	uint32
+// statusComputer computes a transaction status
+type statusComputer struct {
+	selfShardID              uint32
+	uint64ByteSliceConverter typeConverters.Uint64ByteSliceConverter
+	store                    dataRetriever.StorageService
 }
 
-func NewStatusComputer(selfShard uint32) *StatusComputer {
-	statusComputer := &StatusComputer{
-		SelfShard: selfShard}
+// Create a new instance of statusComputer
+func NewStatusComputer(
+	selfShardID uint32,
+	uint64ByteSliceConverter typeConverters.Uint64ByteSliceConverter,
```

Comment:
missing nil checks for `uint64ByteSliceConverter` and `store`.

Reply:
Done!